### PR TITLE
[Priority!] amethyst 0.10 port

### PR DESCRIPTION
**Pull request by KuSpa**
* [x] change cgmath to nalgebra
* [x] make it compile
* [x] change Rotation angles to Rad instead of Deg

### CODE REVIEW

File: src/level/level_state.rs [hunk 1]
```diff
@@ -109,12 +109,17 @@ impl LevelState {
...
+
+        let (screen_w, screen_h) = {
+            let dims = world.read_resource::<ScreenDimensions>();
+            (dims.width(), dims.height())
+        };
 
         world
             .create_entity()
-            .with(Camera::from(Projection::perspective(1.0, Deg(60.0))))
+            .with(Camera::standard_3d(screen_w, screen_h))
```
Comment (karyon) [inline src/level/level_state.rs]: you sure that's equivalent?

File: src/rockraiders.rs [hunk 2]
```diff
@@ -39,18 +31,3 @@ impl<'a, 'b> State<CustomGameData<'a, 'b>, StateEvent> for RockRaiders {
...
-    mat.move_global(Vector3::new(-2.0, 5.0, 2.0));
-    mat.yaw_global(Deg(-45.0));
-    mat.pitch_local(Deg(-45.0));
-
-    world
-        .create_entity()
-        .with(Camera::from(Projection::perspective(1.0, Deg(60.0))))
-        .with(mat)
-        .with(GlobalTransform::default())
-        .build();
```
Comment (karyon) [inline src/rockraiders.rs]: why could you remove all of this?

File: src/systems/camera_movement.rs [hunk 3]
```diff
@@ -38,14 +38,14 @@ impl<'a> System<'a> for CameraMovementSystem {
...
-        if !dir.is_zero() {
+        if !(dir.magnitude() == 0.0) {
             for (transform, _) in (&mut transforms, &cams).join() {
                 dir = transform.matrix() * dir;
                 let move_dir = Vector3::new(dir.x, 0.0, dir.z);
 
                 transform.move_along_global(
-                    move_dir,
-                    time.delta_seconds() * ((x.abs() + z.abs()) / 2.0),
+                    Unit::<Vector3<f32>>::new_normalize(move_dir),
```
Comment (karyon) [inline src/systems/camera_movement.rs]: i'd move all of this to where move_dir is initialized

File: src/level/level_state.rs (hunk 1 above)
Comment (KuSpa) [reply src/level/level_state.rs]: yes, i looked up the sourcecode

File: src/rockraiders.rs (hunk 2 above)
Comment (KuSpa) [reply src/rockraiders.rs]: because its the camera of the main state, where no camera is needed so far

**Pull request merged**